docker/swarmkit
# address unassigned task leak when service is removed

Signed-off-by: Dani Louca <dani.louca@docker.com>



**- What I did**

If a task is not yet assigned (for ex: due to `noSuitableNode` ) and its service has been removed, the task stays in the `unassignedTasks` map until a new leader election.
 
**- How I did it**

Before re-adding the task to `unassignedTasks` map, the fix checks if the task has a valid service in the store.

**- How to test it**

steps to repro:
- set daemon.log to debug
- create a service with a node constraint that does not exist:
ex: `docker service create -d --constraint 'node.labels.type == queue' alpine sleep 10000`
- watch the logs, you should see an error 
`no suitable node available for task`
- delete the service and invoke the tick() by creating another service; watch the log and notice the same warning taskID in pops up after service is deleted

**- Description for the changelog**


Avoid a leak when a service with unassigned tasks is deleted
